tasks/service_stop_amq_broker_instance.yml

```diff
@@ -20,18 +20,39 @@
   when:
     - amq_broker_service_start is defined and amq_broker_service_start == true
     - amq_broker_path_st.stat.exists and amq_broker_path_st.stat.isdir is defined
-  become: true
-  become_user: "{{ amq_broker_user | default(amq_broker_default_user) }}"
-  command: "{{ amq_broker_install_dest }}/{{ amq_broker_instance_dir }}/bin/artemis-service stop"
-  register: stop_result
+  block:
+    - name: Stop broker instance via service
+      become: true
+      become_user: "{{ amq_broker_user | default(amq_broker_default_user) }}"
+      command: "{{ amq_broker_install_dest }}/{{ amq_broker_instance_dir }}/bin/artemis-service stop"
+  rescue:
+    - name: Kill JBoss AMQ 7 Broker
+      include: kill_amq_broker_instance.yml
 
 - name: Stop JBoss AMQ 7 Broker default instance
-  when: amq_broker_service_start is defined and amq_broker_service_start == false
-  become: true
-  become_user: "{{ amq_broker_user | default(amq_broker_default_user) }}"
-  command: "{{ amq_broker_install_dest }}/{{ amq_broker_instance_dir }}/bin/artemis stop"
-  register: stop_result
+  when:
+    - amq_broker_service_start is defined and amq_broker_service_start == false
+    - amq_broker_path_st.stat.exists and amq_broker_path_st.stat.isdir is defined
+  block:
+    - name: Stop broker instance
+      become: true
+      become_user: "{{ amq_broker_user | default(amq_broker_default_user) }}"
+      command: "{{ amq_broker_install_dest }}/{{ amq_broker_instance_dir }}/bin/artemis stop"
+  rescue:
+    - name: Kill JBoss AMQ 7 Broker
+      include: kill_amq_broker_instance.yml
 
-- name: Kill JBoss AMQ 7 Broker
-  when: stop_result | failed and amq_broker_force_kill is defined and amq_broker_force_kill == true
-  include: kill_amq_broker_instance.yml
+- name: Make sure no Artemis/AMQ 7 broker is running on machine
+  when: amq_broker_force_kill is defined and amq_broker_force_kill == true
+  block:
+    - name: Check running artemis process
+      command: 'pgrep -f "java.*artemis"'
+      register: bad_artemis_pid
+      msg: "Bad artemis= {{ bad_artemis_pid }}"
```

Comment:
A better error message must be provided here. "Bad artemis=<pid>" does not provide any clue about what happen. Please provide a message that explains what happened and what should be done to resolve the issue.

Reply:
Also, I'd recommend renaming bad_artemis_pid to something that properly represent its intent/content (ie.: broker_stuck_pid, etc).

Reply:
Agree, I will update it.